Sources/Datadog/Core/System/NetworkPathMonitor.swift

```diff
@@ -7,25 +7,34 @@
 import Network
 
 @available(iOS 12, *)
-extension NWPathMonitor {
+extension NWPathMonitor: NetworkConnectionInfoProviderType {
     var current: NetworkConnectionInfo {
         let info = currentPath
 
-        let isCurrentPathConstrained: Bool? = {
-            if #available(iOS 13.0, *) {
-                return info.isConstrained
-            } else {
-                return nil
-            }
-        }()
+        // NOTE: RUMM-448
+        // We experienced a crash on `info.supportsIPv6` during the app transition from background to foreground.
+        //
+        // By looking at the `NWPath` implementation in `Darwin/Network`:
+        // https://github.com/apple/swift/blob/5c4bbf3de96b733154229a5cfdf16096bbdfdb51/stdlib/public/Darwin/Network/NWPath.swift#L174-L198
+        // we suspect that `nw_path_*` functions might behave unreliable for some `nw_path_get_status()` values.
+        //
+        // This is why we resolve `currentPath` details only on the status values that we were able to
+        // stress test with heavy thread concurrency on real device (`.staisfied` & `.unsatisfied`).
+        let isCurrentPathReliable = info.status != .requiresConnection
```

Comment:
I couldn't understand how `isCurrentPathReliable` variable may help us

Btw, source code of `NWPathMonitor` is very useful for understanding, thanks for finding it 🙏 

Source code shows that once we call `start` with a `queue`, `NWPathMonitor` sets its `currentPath` occassionally within `updateHandler` in `queue` while we  read `currentPath` from other queues.

What we need to do is 
1. the queue is serial (which is already the case ✅ )
2. ensuring that we always read `currentPath` from the same queue with writer queue ❌ 

```swift
class NetworkMonitor {
  private(set) var current: NetworkConnectionInfo?

  init(monitor: NWPathMonitor) {
    ...
    provider.pathUpdateHandler = { [weak self] path in
      // don't read currentPath outside of pathUpdateHandler
      self.current = path.current
    }
    ...
  }
```